.github/workflows/main.yaml

```diff
@@ -0,0 +1,312 @@
+name: Build and Deploy Image
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Define the tag for the code'     
+        required: false
+      branch:
+        description: 'Define branch name'     
+        required: false
+        default: 'main'
+  push:
+    branches:
+      - workflow
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  GITHUB_IMAGE_REPO: ghcr.io/bcgov/dts-endorser-service/
+  OPENSHIFT_IMAGE_REPO: image-registry.apps.silver.devops.gov.bc.ca/4a9599-tools/
+  APP_NAMES: aries-endorser-agent,aries-endorser-db,aries-endorser-backup,aries-endorser-proxy,aries-endorser-api
+
+jobs:
+  build:
+    if: (github.repository == 'bcgov/dts-endorser-service') || (github.event_name == 'workflow_dispatch')
+    name: Build Image
+    permissions:
+      packages: write
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - service: aries-endorser-agent
+            GIT_REPO_URL: hyperledger/aries-endorser-service 
+            DOCKER_FILE_PATH: Dockerfile.acapy # The docker path, file, is the relative path to the docker file from the root of the repo.
+            SOURCE_CONTEXT_DIR: docker/acapy   # The context dir, context, sets the context for the build. i.e. where the build will source files from 
+          - service: aries-endorser-db
+            GIT_REPO_URL: hyperledger/aries-endorser-service
+            PATH:  docker/wallet/config
+            BUILDER_IMAGE: 'quay.io/fedora/postgresql-13:13'
+          - service: aries-endorser-backup
+            GIT_REPO_URL: BCDevOps/backup-container
+            DOCKER_FILE_PATH: Dockerfile  # The docker path, file, is the relative path to the docker file from the root of the repo.
+            SOURCE_CONTEXT_DIR: docker    # The context dir, context, sets the context for the build. i.e. where the build will source files from
+          - service: aries-endorser-proxy
+            DOCKER_FILE_PATH: Dockerfile  # The docker path, file, is the relative path to the docker file from the root of the repo.
+            SOURCE_CONTEXT_DIR: proxy     # The context dir, context, sets the context for the build. i.e. where the build will source files from
+            SOURCE_IMAGE_REGISTRY: "artifacts.developer.gov.bc.ca/docker-remote/"
+            SOURCE_IMAGE_NAME: caddy
+            SOURCE_IMAGE_TAG: latest
+          - service: aries-endorser-api
+            GIT_REPO_URL: hyperledger/aries-endorser-service
+            DOCKER_FILE_PATH: Dockerfile.endorser # The docker path, file, is the relative path to the docker file from the root of the repo.
+            SOURCE_CONTEXT_DIR: endorser # The context dir, context, sets the context for the build. i.e. where the build will source files from
+            BASE_SOURCE_IMAGE_REGISTRY: artifacts.developer.gov.bc.ca/docker-remote/
+            BASE_SOURCE_IMAGE_NAME: python
+            BASE_SOURCE_IMAGE_TAG: 3.10-slim-buster
```

Comment:
Please be consistent with specifying base image variables; SOURCE_IMAGE_REGISTRY, SOURCE_IMAGE_NAME, and SOURCE_IMAGE_TAG.